### PR TITLE
Dt 628 JWT and API changes

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: helm_deploy/values-dev.yaml [hunk 1]
```diff
@@ -28,9 +28,4 @@ env:
   PRISONTONHS_ONLY_PRISONS: "MDI,LEI,PVI,BXI"
 
 whitelist:
-  office: "217.33.148.210/32"
```
Comment (Mjwillis) [inline helm_deploy/values-dev.yaml]: Don't want to whitelist Dev as want TPP to test connectivity

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/config/AuthAwareTokenConverter.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package uk.gov.justice.digital.hmpps.prisontonhs.config
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/config/AuthAwareTokenConverter.kt]: Copied from @andymarke code

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/RestResponsePage.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.databind.JsonNode
+import org.springframework.data.domain.PageImpl
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Pageable
+import java.util.*
+
+
+internal class RestResponsePage<T> : PageImpl<T> {
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/RestResponsePage.kt]: This is done to allow de serialisation of page object

Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/RestResponsePage.kt]: If you were using Jackson you could have used `com.fasterxml.jackson.module:jackson-module-kotlin` to create constructors I think

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/OffenderService.kt [hunk 4]
```diff
@@ -13,75 +13,75 @@ import java.time.Duration
...
   fun getOffender(offenderNo: String): PrisonerStatus? {
     return webClient
-            .get()
-            .uri("$baseUri/api/prisoners/$offenderNo/full-status")
-            .attributes(ServletOAuth2AuthorizedClientExchangeFilterFunction.clientRegistrationId("nomis-api"))
-            .retrieve()
-            .bodyToMono(PrisonerStatus::class.java)
-            .block(offenderTimeout);
+        .get()
+        .uri("$baseUri/api/prisoners/$offenderNo/full-status")
```
[the remark this answers is not in the document]
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/services/OffenderService.kt]: Yes I saw that, think I'll do it the way you have in the next commit.

**Pull request merged**